proposals/function-pointers.md

```diff
@@ -63,174 +64,173 @@ unsafe class Example {
 }
 ```
 
-A `func*` type is a pointer type which means it has all of the capabilities and restrictions of a standard pointer
+A `delegate*` type is a pointer type which means it has all of the capabilities and restrictions of a standard pointer
 type:
+
 - Only valid in an `unsafe` context.
-- Methods which contain a `func*` parameter or return type can only be called from an `unsafe` context.
+- Methods which contain a `delegate*` parameter or return type can only be called from an `unsafe` context.
 - Cannot be converted to `object`.
 - Cannot be used as a generic argument.
-- Can implicitly convert `func*` to `void*`.
-- Can explicitly convert from `void*` to `func*`.
+- Can implicitly convert `delegate*` to `void*`.
+- Can explicitly convert from `void*` to `delegate*`.
 
 Restrictions:
-- Custom attributes cannot be applied to a `func*` or any of its elements.
-- A `func*` parameter cannot be marked as `params`
-- A `func*` type has all of the restrictions of a normal pointer type.
+
+- Custom attributes cannot be applied to a `delegate*` or any of its elements.
+- A `delegate*` parameter cannot be marked as `params`
+- A `delegate*` type has all of the restrictions of a normal pointer type.
 
 ### Function pointer syntax
+
 The full function pointer syntax is represented by the following grammar:
 
-``` 
-funcptr_type = 
-    'func' '*' [calling_convention] type method_arglist |
-    '(' funcptr_type ')' ;
-
-calling_convention = 
-    'cdecl' |
-    'managed' |
-    'stdcall' | 
-    'thiscall' |
-    'unmanaged' ;
+```antlr
+pointer_type
+    : ...
+    | funcptr_type
+    ;
+
+funcptr_type
+    : 'delegate' '*' calling_convention? '<' (funcptr_parameter_modifier? type ',')* funcptr_return_modifier? return_type '>'
+    ;
+
+calling_convention
+    : 'cdecl'
+    | 'managed'
```

Comment:
In the original proposal, naming the `managed` calling convention was required for disambiguation in some case. Are there still ambiguous cases with this proposal where it is needed? 

If there are no such cases, should `managed` be dropped from here? It is better to only have one way to write something. Otherwise, people need to create coding convention to codify which one of the multiple ways is the correct one for the given project.

Reply:
It's no longer required, but I'd prefer to keep it in. Just like we have defaults for accessibility modifiers and also allow you to specify them, I think it's good to allow to specify `managed` even if it's the default.